packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp

```diff
@@ -89,21 +91,31 @@ void Trilinos_Util_ReadHpc2Epetra_internal(
   // the warning is easy to fix.
   if(sizeof(int) == sizeof(int_type)) {
     int numGlobalEquations_int, total_nnz_int;
-
-    cnt = fscanf(in_file,"%d",&numGlobalEquations_int);
+#ifdef DEBUG
+    cnt =
+#endif
+    fscanf(in_file,"%d",&numGlobalEquations_int);
     assert(cnt > 0);
```

Comment:
This line is generating the following build error:
```
02:04:28 /eppic/jenkins/cortado_slave/workspace/EMPIRE-build-test-Trilinos-cortado-gnu-debug-openmp-static/Trilinos/packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp:98:12: error: ‘cnt’ was not declared in this scope
02:04:28      assert(cnt > 0);
02:04:28             ^
```

How on earth did this pass PR testing?

Reply:
Also this popped up in a debug build, so perhaps `DEBUG` isn't the right thing to be checking in the `#ifdef`, because it's not getting switched on when `CMAKE_BUILD_TYPE=DEBUG`.

Reply:
Here are Jenkins jobs showing the failure:
*  [EMPIRE-build-test-Trilinos-cortado-gnu-debug-openmp-static #85](https://jenkins-srn.sandia.gov:8443/job/EMPIRE-build-test-Trilinos-cortado-gnu-debug-openmp-static/85//console)
*  [EMPIRE-build-test-Trilinos-cortado-gnu-debug-serial-static #85](https://jenkins-srn.sandia.gov:8443/job/EMPIRE-build-test-Trilinos-cortado-gnu-debug-serial-static/85//console)

Reply:
Also curious:  Why did we not see this in any of our opt builds?  E.g., [EMPIRE-build-test-Trilinos-cortado-gnu-opt-openmp-static #86](https://jenkins-srn.sandia.gov:8443/job/EMPIRE-build-test-Trilinos-cortado-gnu-opt-openmp-static/86//console).

Reply:
@trilinos/framework, do we intentionally not have the @trilinos-autotester running a debug build?  This PR broke all debug builds of Trilinos simply because debug wasn't tested before it was merged.

Reply:
Also, it seems in TriUtils `DEBUG` is getting switched on when `CMAKE_BUILD_TYPE=RELEASE`.  Not sure what's up with that.